src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/issuefinaldecision/esa/EsaIssueFinalDecisionAboutToSubmitHandlerTest.java

```diff
@@ -168,6 +168,41 @@ public void givenANonIssueFinalDecisionEvent_thenReturnFalse() {
         assertFalse(handler.canHandle(ABOUT_TO_SUBMIT, callback));
     }
 
+    @Test
+    public void givenAnIssueFinalDecisionEventRemoveDraftDecisionNotice() {
+        callback.getCaseDetails().getCaseData().getSscsFinalDecisionCaseData().setWriteFinalDecisionPreviewDocument(documentLink);
+        callback.getCaseDetails().getCaseData().getSscsFinalDecisionCaseData().setWriteFinalDecisionAllowedOrRefused("allowed");
+        callback.getCaseDetails().getCaseData().getSscsFinalDecisionCaseData().setWriteFinalDecisionGenerateNotice("no");
+
+        SscsDocument document1 = SscsDocument.builder().value(SscsDocumentDetails.builder().documentType(DRAFT_DECISION_NOTICE.getValue()).build()).build();
```

Comment:
I would also suggest that these builders could be separated, have a default docDetails object, then set documentType as required in this document object.